aslam_cv/aslam_cameras/include/aslam/cameras/implementation/OmniProjection.hpp

```diff
@@ -888,7 +887,7 @@ bool OmniProjection<DISTORTION_T>::estimateTransformation(
     Eigen::Vector3d backProjection;
 
     if (keypointToEuclidean(imagePoint, backProjection)
-        && backProjection[2] > 0.0) {
+        && backProjection.normalized()[2] > std::cos(80.0*M_PI/180.0)) {
```

Comment:
Because at line 898-899 we divide by z to get pseudo-pinhole points and use them with cv::solvePnP at line 931. This is numerically unstable for angles close to 90 degrees (z is close to zero). 80 degrees was chosen experimentally such that it works on all datasets we used.

Reply:
Hm @VladyslavUsenko was quicker...